flow/config.go

```diff
@@ -164,3 +265,48 @@ func ConfigExists() bool {
 	}
 	return !info.IsDir()
 }
+
+var fields = []string{
+	"projectId",
+	"locationId",
+	"keyRingId",
+	"keyId",
+	"keyVersion",
+}
+
+func kmsKeyFromKeyContext(keyContext map[string]string) (cloudkms.Key, error) {
+
+	for _, field := range fields {
+		if val, ok := keyContext[field]; !ok || val == "" {
+			return cloudkms.Key{}, fmt.Errorf("Could not generate KMS key from Context. Invalid value for %s", field)
+		}
+	}
+
+	return cloudkms.Key{
+		ProjectID:  keyContext[fields[0]],
+		LocationID: keyContext[fields[1]],
+		KeyRingID:  keyContext[fields[2]],
+		KeyID:      keyContext[fields[3]],
+		KeyVersion: keyContext[fields[4]],
+	}, nil
+}
+
+func KeyContextFromKMSResourceID(resourceID string) (map[string]string, error) {
+	resourceSplit := strings.Split(resourceID, "/")
+
+	// Should be 10, since we're assuming the format:
+	// projects/<PROJECTID>/locations/<LOCATION>/keyRings/<KEYRINGID>/cryptoKeys/<KEYID>/cryptoKeyVersions/<KEYVERSION>
+	if len(resourceSplit) != 10 {
+		return nil, fmt.Errorf("Malformed resource ID: %s", resourceID)
+	} else if len(fields) != 5 {
+		Exit(1, "list of fields for key context has unexpected length")
+	}
```

Comment:
Maybe just use a regular expression here. This would allow further validation of the non-variable parts.

Here's a Go Playground: https://play.golang.org/p/tuuZtCy-WkV